sklearn/feature_selection/_from_model.py

```diff
@@ -256,6 +256,8 @@ def fit(self, X, y=None, **fit_params):
             raise NotFittedError("Since 'prefit=True', call transform directly")
         self.estimator_ = clone(self.estimator)
         self.estimator_.fit(X, y, **fit_params)
+        if hasattr(self.estimator_, "feature_names_in_"):
```

Comment:
I think it's a bit strange to have different mechanisms for ``n_features_in_`` and ``feature_names_in_``. What was the reason for the property in ``n_features_in_``? I'm sure we had a reason?

Reply:
[SLEP010](https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep010/proposal.html#considerations) allowed for both options:

> In most meta-estimators, the input validation is handled by the sub-estimator(s). The n_features_in_ attribute of the meta-estimator is thus explicitly set to that of the sub-estimator, either via a @property, or directly in fit().

I prefer setting it in `fit` because it is a little more complicated in setting a `@property`. If we got https://github.com/scikit-learn/scikit-learn/pull/19948 merged, then using `@property` feels slightly better.

I recall in a PR review @NicolasHug and @jeremiedbb had a discussion about this.

Reply:
I also prefer the "setting it in fit" route btw

Reply:
I also like the "setting it in fit" better than properties. But no need to change this as part of this PR.

Reply:
> I'm sure we had a reason?

> I recall in a PR review @NicolasHug and @jeremiedbb had a discussion about this.

Indeed, the discussion ends up that there is no reason :) +1 for "setting in fit" and +1 to delegate in another PR.

Reply:
I think the only case where a property is really needed is for e.g. `SparseCoder`, where the attribute is known right after `__init__()`, i.e. before `fit()`.

I think I used a property in meta estimators to mimic the implementation of stuff like `classes_` or `intercept_`. But I don't feel strongly about this.

Reply:
ok sounds good!